src/zed_cpu.cpp

```diff
@@ -0,0 +1,119 @@
+#include <zed_cpu.hpp>
+
+ZedCameraNode::ZedCameraNode(
+  const std::shared_ptr<ros::NodeHandle> & nh,
+  const std::shared_ptr<image_transport::ImageTransport> & it)
+: nh_(nh), it_(it)
+{
+  // ROS initialization
+  left_image_pub_ = it_->advertise("rgb/left_image", 1);
+  right_image_pub_ = it_->advertise("rgb/right_image", 1);
+  imu_pub_ = nh_->advertise<sensor_msgs::Imu>("imu_data", 1);
+
+  cameraInit();
+  sensorInit();
+}
+
+void ZedCameraNode::run()
+{
+  publishImages();
+  publishIMU();
+}
+
+void ZedCameraNode::cameraInit()
+{
+  // Initialize ZED camera
+  sl_oc::video::VideoParams params;
+  params.res = sl_oc::video::RESOLUTION::HD720;
+  params.fps = sl_oc::video::FPS::FPS_60;
+  params.verbose = sl_oc::VERBOSITY::ERROR;
+
+  // Create Video Capture
+  cap_ = std::make_unique<sl_oc::video::VideoCapture>(params);
+  if (!cap_->initializeVideo()) {
+    ROS_ERROR("Cannot open camera video capture");
```

Comment:
```suggestion
    ROS_ERROR("[%s] Cannot open camera video capture", node_name_.c_str());
```
node name you typically assign at the beginning of the constructor:
`node_name_ = ros::this_node::getName();`
Applies for all log messages